### PR TITLE
Update to deal.II 9.2

### PR DESCRIPTION
**Pull request by peterrum**
I'll run the tests later.

### CODE REVIEW

File: source/level_set_okz.cc [hunk 1]
```diff
@@ -77,9 +77,11 @@ namespace AssemblyData
 {
   struct Data
   {
+    Data(){} // TODO?
```
Comment (peterrum) [inline source/level_set_okz.cc:80]: Not sure about this. `Threads::ThreadLocalStorage` seems to need it.

File: source/navier_stokes_preconditioner.cc [hunk 2]
```diff
@@ -1327,6 +1328,29 @@ namespace AssemblyData
...
+                         update_normal_vectors),
+    fe_values_lin_u (fe_values_u.get_mapping(),
+                     fe_values_u.get_fe(),
+                     fe_values_u.get_quadrature(),
+                     fe_values_u.get_update_flags()),
+    fe_values_lin_p (fe_values_p.get_mapping(),
+                     fe_values_p.get_fe(),
+                     fe_values_p.get_quadrature(),
+                     fe_values_p.get_update_flags())
+   {}
```
Comment (peterrum) [inline source/navier_stokes_preconditioner.cc:1353]: same here

File: source/navier_stokes.cc [hunk 3]
```diff
@@ -1300,6 +1300,8 @@ void NavierStokes<dim>::output_solution (const std::string output_name,
   std::vector<DataComponentInterpretation::DataComponentInterpretation>
   vector_component_interpretation
   (dim, DataComponentInterpretation::component_is_part_of_vector);
+  
+  solution.update_ghost_values();
```
Comment (peterrum) [inline source/navier_stokes.cc:1304]: This line was missing.

File: source/level_set_okz.cc [hunk 4]
```diff
@@ -662,11 +666,11 @@ LevelSetOKZSolver<dim>::local_compute_curvature_rhs (const MatrixFree<dim,double
       // and be done before we apply the derivatives, which is done in the
       // code below.
       bool all_zero = true;
-      for (unsigned int i=0; i<normal_values.dofs_per_cell; ++i)
+      for (unsigned int i=0; i<normal_values.dofs_per_component; ++i)
```
Comment (peterrum) [inline source/level_set_okz.cc:669]: I guess this also has change in the mean time.

**Comment by peterrum** [comment]
@kronbichler I have updated the code to deal.II 9.2. My experiences were:
- running test out-of-source is not possible
- running tests takes quite long (I have not waited till completion and instead run everything in paralle with 40 processes and even so `flow_past_cylinder` and `flow_past_square_cylinder` took very long)
- if I am doing something wrong in the last point and the tests are indeed much faster I would suggest to introduce CI
- in some of the tests there are segmentation faults during calling the final destructors - I am not sure about the reason, e.g., `mpirun -np 40 spurious_currents spurious_currents_ls.prm`
- there is not .clang-format file so that I could not format the changes

FYI @mschreter

**Review by kronbichler — APPROVED** [review]

**Comment by kronbichler** [comment]
Let us merge this now; I will add a suitable clang-format file later and look into the tests.

**Comment by kronbichler** [comment]
I did commits to fix those:

> running test out-of-source is not possible (I had to manually copy the input files to the build folder)
> there is not .clang-format file so that I could not format the changes

Regarding the tests: These are known to be slow in debug mode but reasonably fast in release mode. That said, there were some spurious commits in the test folder resulting in too big computations. More of them could surely be somewhat reduced but I did not get around to doing it yet. The more annoying issue is that I get failures due to iteration counts of the iterative solver to be one or two off; that would require some more work. I have to check the MPI issue locally with debug mode later.

**Comment by kronbichler** [comment]
Another todo is that some tests, like the `rising_bubble.cc`, run with threads as well, variable set to -1. This leads to very poor performance on machines with many cores and paralle tests (I just verified that locally).

**Comment by peterrum** [comment]
Thanks @kronbichler!

**Pull request merged**